File: Backend/BillX-Pos-System/src/main/java/com/BillX/Service/AuthService.java
package com.BillX.Service;

import com.BillX.Exception.UserException;
import com.BillX.Payload.dto.UserDto;
import com.BillX.Payload.response.AuthResponse;

public interface AuthService {
    AuthResponse signup(UserDto userDto) throws UserException;
    AuthResponse login(UserDto userDto) throws UserException;
}

File: Frontend/src/services/authService.ts
import api from "../api/axios";
import { AuthResponse, LoginRequest, UserDto } from "../types";

export const authService = {
  login: async (data: LoginRequest) => {
    const res = await api.post<AuthResponse>("/auth/login", data);
    return res.data;
  },
  signup: async (data: { name: string; email: string; password: string }) => {
    const res = await api.post<AuthResponse>("/auth/register", data);
    return res.data;
  },
};
